modules/paginator/paginator.go

```diff
@@ -0,0 +1,206 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+// Copyright 2015 Unknwon. Licensed under the Apache License, Version 2.0
```

Comment:
I'm not a lawyer but I'm not sure if we're allowed to re-license this under MIT-style license(even though we credit it, the source code becomes effectively MIT-licensed source code).

Reply:
What I mean is before the code is under Apache License. Then newly written code is under MIT license, I did some changes to the code.

Since you worry about it, I changed these texts now.